tests/test_client.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2017 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA.
+#
+# Authors:
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import os
+import sys
+import time
+import unittest
+
+import httpretty
+import pkg_resources
+import requests
+
+# Hack to make sure that tests import the right packages
+# due to setuptools behaviour
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), ".."))
+pkg_resources.declare_namespace('perceval.backends')
+
+from perceval.client import HttpClient, RateLimitHandler
+
+
+CLIENT_API_URL = "https://gateway.marvel.com/v1/"
+CLIENT_SPIDERMAN_URL = "https://gateway.marvel.com/v1/public/characters/1"
+CLIENT_SUPERMAN_URL = "https://gateway.marvel.com/v1/public/characters/2"
+CLIENT_BATMAN_URL = "https://gateway.marvel.com/v1/public/characters/3"
+CLIENT_IRONMAN_URL = "https://gateway.marvel.com/v1/public/characters/4"
+
+
+class MockedClient(HttpClient, RateLimitHandler):
+
+    def __init__(self, base_url, default_sleep_time=0.1, max_retries=1):
+
+        super().__init__(base_url, default_sleep_time=default_sleep_time, max_retries=max_retries)
+        super().setup_rate_limit_handler()
+
+
+class TestHttpClient(unittest.TestCase):
+    """Http client tests """
+
+    def test_initialization(self):
```

Comment:
We need another test where we check that the parameters passed to the class are well set. I mean, there aren't assignment errors to variables.